src/services/swap/swap-service.ts

```diff
@@ -0,0 +1,39 @@
+import { SwapServiceError } from "@domain/swap/errors"
+import { wrapAsyncFunctionsToRunInSpan } from "@services/tracing"
+
+import { LoopUtils } from "./swap-utils"
+
+export const SwapService = (): ISwapService => {
+  const loopUtils = LoopUtils()
+  const loopService = loopUtils.getActiveLoopService()
+  const healthCheck = async () => {
+    try {
+      const isServiceUp = await loopService.healthCheck()
+      if (isServiceUp) return true
+      return false
+    } catch (e) {
+      return false
+    }
+  }
+
+  const swapOut = async (amount) => {
```

Comment:
imo this function should get both `{ amount, swapDestAddress }` as parameters, instead of solely relying on `amount`.

swapDestAddress should be injected at the app level.

this would remove the dependency from this service to the `OnChainService`